app/models/content_block.rb

```diff
@@ -1,3 +1,106 @@
+# coding: utf-8
+
 class ContentBlock < ActiveRecord::Base
-  include Hyrax::ContentBlockBehavior
+  # The keys in this registry are "public" names for collaborator
+  # objects, and the values are reserved names of ContentBlock
+  # instances, which Hyrax uses as identifiers. Values also correspond
+  # to names of messages that can be sent to the ContentBlock class to
+  # return defined ContentBlock instances.
+  NAME_REGISTRY = {
+    marketing: :marketing_text,
+    researcher: :featured_researcher,
+    announcement: :announcement_text,
+    about: :about_page,
+    help: :help_page,
+    terms: :terms_page,
+    agreement: :agreement_page
+  }.freeze
+
+  # NOTE: method defined outside the metaclass wrapper below because
+  # `for` is a reserved word in Ruby.
+  def self.for(key)
+    key = key.respond_to?(:to_sym) ? key.to_sym : key
+    return new unless whitelisted?(key)
```

Comment:
Why are we returning new here?  Woudn't raising an ArgumentError be the right thing here?